src/common/protocol.ts

```diff
@@ -150,6 +150,54 @@ export class Protocol {
 		}
 	}
 
+	toString(): string {
+		// based on Uri scheme for SSH https://tools.ietf.org/id/draft-salowey-secsh-uri-00.html#anchor1 and heuristics of how GitHub handles ssh url
```

Comment:
@queerviolet may I have your expertise and review on this, my heuristics of how GitHub generates ssh uri for repos might be wrong.

Reply:
This looks right, to the best of my knowledge. +1 adding some tests on it. 👍🏽